jupyterhub_traefik_proxy/etcd.py

```diff
@@ -280,18 +221,14 @@ def _stop_traefik(self):
             return
 
         target = value.decode()
-        backend_alias = traefik_utils.create_backend_alias_from_url(target)
-        frontend_alias = traefik_utils.create_frontend_alias_from_url(target)
-        backend_url_path = traefik_utils.create_backend_url_path(self, backend_alias)
-        backend_weight_path = traefik_utils.create_backend_weight_path(
-            self, backend_alias
-        )
-        frontend_backend_path = traefik_utils.create_frontend_backend_path(
-            self, frontend_alias
-        )
-        frontend_rule_path = traefik_utils.create_frontend_rule_path(
-            self, frontend_alias
-        )
+        (
+            backend_alias,
+            backend_url_path,
+            backend_weight_path,
+            frontend_alias,
+            frontend_backend_path,
+            frontend_rule_path,
+        ) = traefik_utils.generate_route_keys(self, target, routespec)
```

Comment:
Maybe for later: returning long tuples can get unwieldy and hard to follow. At this point, it might be worth using a [namedtuple](https://docs.python.org/3/library/collections.html#collections.namedtuple) to return a single object with named attribute access.